pkg/oci/oci.go

```diff
@@ -259,16 +268,23 @@ func listGadgetImages(ctx context.Context, store *oci.Store) ([]*GadgetImageDesc
 	images := []*GadgetImageDesc{}
 	err := store.Tags(ctx, "", func(tags []string) error {
 		for _, fullTag := range tags {
-			repository, err := getRepositoryFromImage(fullTag)
+			parsed, err := reference.Parse(fullTag)
 			if err != nil {
-				log.Debugf("getting repository from image %q: %s", fullTag, err)
+				log.Debugf("parsing image %q: %s", fullTag, err)
 				continue
 			}
-			tag, err := getTagFromImage(fullTag)
-			if err != nil {
-				log.Debugf("getting tag from image %q: %s", fullTag, err)
-				continue
+
+			var repository string
+			if named, ok := parsed.(reference.Named); ok {
+				repository = named.Name()
+			}
+			repository = strings.TrimPrefix(repository, defaultDomain+"/"+officialRepoPrefix)
```

Comment:
nit: Perhaps we can trim the `defaultDomain` for cases like `ghcr.io/foo/bar` to be `foo/bar`

Reply:
I prefer to trim it only when the full thing matches. I'm don't think we should be treating any other think than "ghcr.io/inspektor-gadget/gadgets" in a special way.